### PR TITLE
[Improvement] Implement design theme grid on ui component

### PR DESCRIPTION
**Pull request by simpleadm**
### Description
Several versions ago backend grid widgets (`\Magento\Backend\Block\Widget\Grid`) were marked as 
deprecated in favour of UI component implementation. This PR refactors design theme grid to use UI components. 
Grid location: Content > Design > Themes

At least it will save time of third party developers who needs to extend themes grid :)

### Manual testing scenarios
Before changes:
image

After changes:
image

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by VladimirZaets** [comment]
Hi @simpleadm , any updates?

**Comment by simpleadm** [comment]
Hi @VladimirZaets ,
what updates you are waiting for?

**Comment by VladimirZaets** [comment]
Hi @simpleadm , we have 2 functional tests that were failed with your changes. That's are  `Magento\Widget\Test\TestCase\CreateWidgetsEntityTest.test with data set "CreateWidgetEntityTestWithDifferentThemes_0"`
and `Magento\Widget\Test\TestCase\DeleteWidgetEntityTest.test with data set "DeleteWidgetEntityTestVariation1_0"`. 
Can you look and fix them, please?
If you need some help or guide how to running functional tests locally, please let us know.

**Comment by VladimirZaets** [comment]
@magento-engcom-team give me new test instance

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets. Thank you for your request. I'm working on Magento instance for you

**Comment by magento-engcom-team** [comment]
Hi @VladimirZaets, here is your [new Magento instance](https://pr-14470.engcom.dev.magento.com).
Admin access: [https://pr-14470.engcom.dev.magento.com/admin](https://pr-14470.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Please make sure you are PR **author** or **assignee** to access the instance.

**Comment by simpleadm** [comment]
Hi @VladimirZaets ,
Please check updates.

File: app/code/Magento/Theme/Model/ResourceModel/Theme/Collection.php [hunk 1]
```diff
@@ -17,6 +17,11 @@ class Collection extends \Magento\Framework\Model\ResourceModel\Db\Collection\Ab
      */
     const DEFAULT_PAGE_SIZE = 6;
 
+    /**
+     * @var string
+     */
+    protected $_idFieldName = 'theme_id';
```
Comment (VladimirZaets) [inline app/code/Magento/Theme/Model/ResourceModel/Theme/Collection.php]: Please change protected to private due to Magento [Technical Guidelines](https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html)

File: app/code/Magento/Theme/Model/ResourceModel/Theme/Grid/Collection.php [hunk 2]
```diff
@@ -5,11 +5,62 @@
...
 /**
  * Theme grid collection
  */
-class Collection extends \Magento\Theme\Model\ResourceModel\Theme\Collection
+class Collection extends ThemeCollection implements SearchResultInterface
 {
+    /**
+     * @var AggregationInterface
+     */
+    protected $aggregations;
```
Comment (VladimirZaets) [inline app/code/Magento/Theme/Model/ResourceModel/Theme/Grid/Collection.php]: Please change protected to private due to Magento [Technical Guidelines](https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html)

File: app/code/Magento/Theme/Model/ResourceModel/Theme/Grid/Collection.php [hunk 3]
```diff
@@ -5,11 +5,62 @@
...
 
+use Magento\Theme\Model\ResourceModel\Theme\Collection as ThemeCollection;
+use Magento\Framework\Api\Search\SearchResultInterface;
+use Magento\Framework\Api\Search\AggregationInterface;
+
 /**
  * Theme grid collection
  */
-class Collection extends \Magento\Theme\Model\ResourceModel\Theme\Collection
+class Collection extends ThemeCollection implements SearchResultInterface
```
Comment (VladimirZaets) [inline app/code/Magento/Theme/Model/ResourceModel/Theme/Grid/Collection.php]: Hi @simpleadm, please make some refactoring:
1) The collection shouldn't extend the ThemeCollection collection, it should only implement SearchResultInterface.
2) Use ThemeCollection to getting access to data inside implemented methods.

**Comment by simpleadm** [comment]
Hi @VladimirZaets ,
Please check new changes.

**Comment by VladimirZaets** [comment]
Hi @simpleadm, thanks, I took this PR in progress.

**Comment by VladimirZaets** [comment]
Hi @simpleadm. The grid rows are not clickable, consequently, edit page is not available too.
Please fix it. Thank

**Comment by simpleadm** [comment]
Hi @VladimirZaets ,
View action was added, please check.

**Comment by VladimirZaets** [comment]
Hi @simpleadm. 
We tried to test your fix and found the next problem:
View action redirects to Dashboard when Add Secret Key to URLs=Yes

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=14470) <br/>All committers have signed the CLA.

**Comment by simpleadm** [comment]
Hi @VladimirZaets , 
Url was fixed.

**Comment by magento-engcom-team** [comment]
Hi @simpleadm. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.0. 
Please check the release notes for final confirmation.

**Pull request merged**